Sources/TensorFlow/Operators/NN.swift

```diff
@@ -293,279 +406,148 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
-                        Int32(kernelSize.3), Int32(kernelSize.4)],
-                strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3),
-                          Int32(strides.4)],
-                padding: padding.raw
-            )
-        })
-    }
+@differentiable(wrt: (x, filter), vjp: _vjpConv2D)
+func conv2D<Scalar: TensorFlowFloatingPoint>(
+    _ x: Tensor<Scalar>,
+    filter: Tensor<Scalar>,
+    strides: (Int, Int, Int, Int),
+    padding: Padding
+    ) -> Tensor<Scalar> {
+    return Raw.conv2D(
+        x,
+        filter: filter,
+        strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3)],
+        padding: padding.raw2,
+        explicitPaddings: [])
 }
 
-public extension Tensor where Scalar: FloatingPoint {
-    /// Computes a 2-D convolution using `self` as input, with the specified
-    /// filter, strides, and padding.
-    ///
-    /// - Parameters:
-    ///     - filter: The convolution filter.
-    ///     - strides: The strides of the sliding filter for each dimension of the
-    ///         input.
-    ///     - padding: The padding for the operation.
-    /// - Precondition: `self` must have rank 4.
-    /// - Precondition: `filter` must have rank 4.
-    @inlinable
-    @differentiable(
-        wrt: (self, filter),
-        vjp: _vjpConvolved2D where Scalar: TensorFlowFloatingPoint)
-    func convolved2D(
-        withFilter filter: Tensor,
-        strides: (Int, Int, Int, Int),
-        padding: Padding
-    ) -> Tensor {
-        return Raw.conv2D(
-            self,
-            filter: filter,
-            strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2), Int32(strides.3)],
-            padding: padding.raw2,
-            explicitPaddings: [])
-    }
-
-    /// Computes a 3-D convolution using `self` as input, with the specified
-    /// filter, strides, and padding.
-    ///
-    /// - Parameters:
-    ///     - filter: The convolution filter.
-    ///     - strides: The strides of the sliding filter for each dimension of the
-    ///         input.
-    ///     - padding: The padding for the operation.
-    /// - Precondition: `self` must have rank 5.
-    /// - Precondition: `filter` must have rank 5.
-    @inlinable @inline(__always)
-    @differentiable(
-        wrt: (self, filter), vjp: _vjpConvolved3D
-        where Scalar: TensorFlowFloatingPoint
-    )
-    func convolved3D(
-        withFilter filter: Tensor,
-        strides: (Int, Int, Int, Int, Int),
-        padding: Padding
-    ) -> Tensor {
-        return Raw.conv3D(
-            self,
-            filter: filter,
-            strides: [Int32(strides.0), Int32(strides.1), Int32(strides.2),
-                      Int32(strides.3), Int32(strides.4)],
-            padding: padding.raw)
-    }
+/// Computes a 3-D convolution using `self` as input, with the specified
+/// filter, strides, and padding.
+///
+/// - Parameters:
+///     - filter: The convolution filter.
```

Comment:
Reduce indentation. The bullet for a detailed comment item should start form the indentation of the text on the previous line.

```suggestion
///   - filter: The convolution filter.
```